[AMDGPU][MC] Allow dpp in v_dot2_f32_bf16 for GFX11 and 12

jwanggit86 opened the pull request:
Allowing the dpp operand in v_dot2_f32_bf16 for GFX11 and 12.

jwanggit86 committed:
[AMDGPU][MC] Allow dpp in v_dot2_f32_bf16 for GFX11 and 12

Allowing the dpp operand in v_dot2_f32_bf16 for GFX11 and 12.

rampitec commented on llvm/lib/Target/AMDGPU/VOP3PInstructions.td:
```diff
@@ -1878,6 +1878,8 @@ defm V_DOT4_F32_BF8_FP8 : VOP3P_Realtriple<GFX12Gen, 0x25>;
 defm V_DOT4_F32_FP8_FP8 : VOP3P_Realtriple<GFX12Gen, 0x26>;
 defm V_DOT4_F32_BF8_BF8 : VOP3P_Realtriple<GFX12Gen, 0x27>;
 
+defm V_DOT2_F32_BF16 : VOP3P_Realtriple<GFX12Gen, 0x1a>;
```
There is VOP3P_Realtriple_gfx11_gfx12 already.

jwanggit86 replied:
Where is it? I couldn't find it.

rampitec replied:
Oh, I see. Take it from downstream, added 2 weeks ago by Mirko.

jwanggit86 replied:
Included in latest commit.

rampitec commented on llvm/test/MC/AMDGPU/gfx11_asm_vop3p.s:
```diff
@@ -45,6 +45,15 @@ v_dot2_f32_bf16 v5, src_scc, vcc_lo, src_scc neg_lo:[1,0,0] neg_hi:[1,0,0]
 v_dot2_f32_bf16 v255, 0xfe0b, vcc_hi, 0.5 neg_lo:[0,1,0] neg_hi:[0,1,0] clamp
 // GFX11: [0xff,0xc2,0x1a,0xcc,0xff,0xd6,0xc0,0x5b,0x0b,0xfe,0x00,0x00]
 
+v_dot2_f32_bf16_e64_dpp v1, v2, v3, v4 quad_perm:[0,1,2,3] row_mask:0xf bank_mask:0xf
```
Should go into gfx11_asm_vop3p_dpp16.s.

jwanggit86 replied:
done.

rampitec commented on llvm/test/MC/AMDGPU/gfx12_asm_vop3p.s:
```diff
@@ -45,6 +45,15 @@ v_dot2_f32_bf16 v5, src_scc, vcc_lo, src_scc neg_lo:[1,0,0] neg_hi:[1,0,0]
 v_dot2_f32_bf16 v255, 0xfe0b, vcc_hi, 0.5 neg_lo:[0,0,0] neg_hi:[0,0,0] clamp
 // GFX12: [0xff,0xc0,0x1a,0xcc,0xff,0xd6,0xc0,0x1b,0x0b,0xfe,0x00,0x00]
 
+v_dot2_f32_bf16_e64_dpp v1, v2, v3, v4 quad_perm:[0,1,2,3] row_mask:0xf bank_mask:0xf
```
Should go into gfx12_asm_vop3p_dpp16.s.

jwanggit86 replied:
done.

rampitec reviewed (commented):
Missing dpp8 and disasm tests.

jwanggit86 committed:
This commit:

(1) created VOP3P_Realtriple_gfx11_gfx12 (2) updated asm tests
(3) created disasm tests.

rampitec reviewed (approved):
LGTM

jwanggit86 committed:
Empty commit to trigger build
